Typescript typing issue with work and batchSize

Hello @timgit,

There is an issue with the typing when using `boss.work` with `batchSize`.

As the documentations say, and as the tests files show, when using batchSize in work options, we should provide a work handler that accept an array of jobs as argument.

But when doing this in TS, there is an error.

```typescript
await boss.work(
    'myqueue',
    { batchSize: 3 },
    (jobs: Job<Foo>[]) => Promise.resolve() 
);
```

```
error TS2769: No overload matches this call.
  The last overload gave the following error.
    Argument of type '(jobs: Job<Foo>[]) => Promise<void>' is not assignable to parameter of type 'WorkHandler<unknown>'.
      Types of parameters 'jobs' and 'job' are incompatible.
        Type 'Job<unknown>' is missing the following properties from type 'Job<Foo>[]': length, pop, push, concat, and 29 more.
```

The typing misses some overloads to handle this case.

I prepare a PR.

Cheers.

--- a/types.d.ts
+++ b/types.d.ts
@@ -107,16 +107,23 @@ declare namespace PgBoss {
     newJobCheckIntervalSeconds?: number;
   }
 
-  interface JobFetchOptions {
-    teamSize?: number;
-    teamConcurrency?: number;
-    teamRefill?: boolean;
-    batchSize?: number;
+  interface CommonJobFetchOptions {
     includeMetadata?: boolean;
     enforceSingletonQueueActiveLimit?: boolean;
   }
 
+  type JobFetchOptions  = CommonJobFetchOptions & {
+    teamSize?: number;
+    teamConcurrency?: number;
+    teamRefill?: boolean;
+  }
+
+  type BatchJobFetchOptions = CommonJobFetchOptions & {
+    batchSize: number;
+  }
+
   type WorkOptions = JobFetchOptions & JobPollingOptions
+  type BatchWorkOptions = BatchJobFetchOptions & JobPollingOptions
 
   type FetchOptions = {
     includeMetadata?: boolean;
@@ -127,8 +134,16 @@ declare namespace PgBoss {
     (job: PgBoss.Job<ReqData>): Promise<void>;
   }
 
+  interface BatchWorkHandler<ReqData> {
+    (job: PgBoss.Job<ReqData>[]): Promise<void>;
+  }
+
   interface WorkWithMetadataHandler<ReqData> {
     (job: PgBoss.JobWithMetadata<ReqData>): Promise<void>;
+  }
+
+  interface BatchWorkWithMetadataHandler<ReqData> {
+    (job: PgBoss.JobWithMetadata<ReqData>[]): Promise<void>;
   }
 
   interface Request {
@@ -301,6 +316,9 @@ declare class PgBoss extends EventEmitter {
   work<ReqData>(name: string, handler: PgBoss.WorkHandler<ReqData>): Promise<string>;
   work<ReqData>(name: string, options: PgBoss.WorkOptions & { includeMetadata: true }, handler: PgBoss.WorkWithMetadataHandler<ReqData>): Promise<string>;
   work<ReqData>(name: string, options: PgBoss.WorkOptions, handler: PgBoss.WorkHandler<ReqData>): Promise<string>;
+
+  work<ReqData>(name: string, options: PgBoss.BatchWorkOptions & { includeMetadata: true }, handler: PgBoss.BatchWorkWithMetadataHandler<ReqData>): Promise<string>;
+  work<ReqData>(name: string, options: PgBoss.BatchWorkOptions, handler: PgBoss.BatchWorkHandler<ReqData>): Promise<string>;
 
   onComplete(name: string, handler: Function): Promise<string>;
   onComplete(name: string, options: PgBoss.WorkOptions, handler: Function): Promise<string>;

